tests/core_tests.py

```diff
@@ -157,6 +172,39 @@ def test_slices(self):
             logging.info("[{name}]/[{method}]: {url}".format(**locals()))
             self.client.get(url)
 
+    def test_add_slice(self):
+        self.login(username='admin')
+
+        # Click on the + to add a slice
+        url = '/slicemodelview/add'
+        resp = self.get_resp(url)
+        assert 'Click on a' in resp
+        assert 'to create a Slice' in resp
+
+        tables = db.session.query(models.SqlaTable).all()
+        for table in tables:
+            # Check that the /tablemodelview/list view contains each table and
+            # their show, edit, delete buttons
+            assert table.name in resp
+            assert '/superset/explore/table/{}'.format(table.id) in resp
+            assert '/tablemodelview/show/{}'.format(table.id) in resp
+            assert '/tablemodelview/edit/{}'.format(table.id) in resp
+            assert '/tablemodelview/delete/{}'.format(table.id) in resp
+
+            # Visit each table's explore view and make sure some
+            # important functionality is there
+            url = 'superset/explore/table/{}/'.format(table.id)
+            explore_view = self.get_resp(url)
+            assert '[{}] - untitled'.format(table.name) in explore_view
+            assert 'Datasource & Chart Type' in explore_view
+            assert 'Save as' in explore_view
+            assert 'Query' in explore_view
+
+        assert not '/explore/table/{}'.format(len(tables) + 1) in resp
```

Comment:
I was testing that exactly the right number of tables appear on the page. If you imagine the case where all the correct tables appear plus extraneous tables, this test would catch that case. Do you think this is going overboard?